src/rebar_config.erl

```diff
@@ -92,19 +91,24 @@ consult_lock_file(File) ->
 %% at most once.
 %% The warning can also be cancelled by configuring the `warn_config_vsn'
 %% OTP env variable.
--spec warn_vsn_once() -> ok.
-warn_vsn_once() ->
+-spec warn_vsn_once(string()) -> ok.
+warn_vsn_once(Vsn) ->
     Warn = application:get_env(rebar, warn_config_vsn) =/= {ok, false},
     application:set_env(rebar, warn_config_vsn, false),
     case Warn of
         false -> ok;
         true ->
-            ?WARN("Rebar3 detected a lock file from a newer version. "
-                  "It will be loaded in compatibility mode, but important "
-                  "information may be missing or lost. It is recommended to "
-                  "upgrade Rebar3.", [])
+            warning_for_vsn(Vsn)
     end.
 
+warning_for_vsn([Maj, _dot,  Min, _dot, _Patch]) when Maj =:= $1 andalso Min =< $1 ->
+    ?WARN("Rebar3 detected a lock file incompatible with this version of rebar3. "
```

Comment:
It might be worth giving more explanation here. That is, to specifically mention a checksum incompatibility, but not sure if most users will care about that detail.

Reply:
That warning is a sign that we're fully breaking backwards compat. See my prior comment about the attribute names so that we may avoid that.